idaes/core/util/model_statistics.py

```diff
@@ -377,7 +378,32 @@ def number_activated_equalities(block):
     Returns:
         Number of activated equality Constraint components in block
     """
-    return sum(1 for _ in activated_equalities_generator(block))
+    return sum(
+        1 for _ in activated_equalities_generator(block)
+    ) + number_grey_box_equalities(block)
+
+
+def number_grey_box_equalities(block) -> int:
+    """
+    Function to compute total number of equality constraints for all GreyBox objects in this block.
+
+    A GreyBox model is always assumed to be 0DOFs where each output[i]==f(inputs)
+    where f is GreyBox model, this should be true regardless if
+    GreyBox model is doing internal optimization or not, as every output
+    is calculated through a the GreyBox internal model using provided inputs.
+
+    Args:
+        block : pyomo concrete model or pyomo block
+
+    Returns:
+        Number of equality constraints in all GreyBox objects on the provided block
+    """
+    equalities = 0
+    for grey_box in _iter_indexed_block_data_objects(
+        block, ctype=ExternalGreyBoxBlock, active=True, descend_into=True
+    ):
+        equalities += len(grey_box.outputs)
+    return equalities
```

Comment:
As written, this will be incorrect for grey box models that have a nonzero number of "explicitly defined equality constraints" (as opposed to the equality constraints that are created to define the outputs). I think all that's needed to correct this is:
```python
equalities += grey_box.get_external_model().n_equality_constraints()
```

Reply:
Right, did not think about this scenario! Updated to test graybox with suggestion, this now includes an example constraint that binds 2 inputs together. 

```    class BasicGrayBox(ExternalGreyBoxModel):
        def input_names(self):
            return ["a1", "a2", "a3"]

        def output_names(self):
            return ["o1", "o2"]

        def equality_constraint_names(self):
            return ["a12"]

        def evaluate_equality_constraints(self):
            a1 = self._input_values[0]
            a2 = self._input_values[1]
            return [a1 * 0.5 - a2]